project/script_TP.tcl

```diff
@@ -11,24 +11,25 @@ open_project -reset trackletProcessor
 
 # source files
 set CFLAGS {-std=c++11 -I../TrackletAlgorithm}
-set_top TrackletProcessor_L1L2D
+set_top TrackletProcessor_L3L4C
 add_files ../TrackletAlgorithm/TrackletProcessor.cc -cflags "$CFLAGS"
-add_files -tb ../TestBenches/TrackletProcessor_L1L2D_test.cpp -cflags "$CFLAGS"
+add_files -tb ../TestBenches/TrackletProcessor_L3L4C_test.cpp -cflags "$CFLAGS"
 
 open_solution "solution1"
 
 # Define FPGA, clock frequency & common HLS settings.
 source settings_hls.tcl
+set_clock_uncertainty .05
```

Comment:
@jasonfan393 Is this really necessary? Given how unreliable the timing estimates from HLS are, this doesn't seem realistic.

Reply:
as of now, the L3L4 TP comes pretty close to the 4ns threshold, I needed to set the uncertainty this low to actually have it pass csynth. This has been an issue I've been looking at, but don't really know where that can be improved right now.

Reply:
Does changing the uncertainty actually change the timing estimate, i.e., the minimum clock period, or resource utilization that HLS reports, or does it just change whether it says it passes?

It sounds like it's the latter, in which case, you should comment out this line. We don't care too much about the timing estimate from HLS, since it's so unreliable and the estimate after implementation is presumably more accurate. And other modules, like the TE, fail timing according to HLS but pass according to the post-implementation estimate.

Reply:
yes, it's the latter, just commented it out.